File: Examples/Features/Worksheets/UnprotectWorksheet.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace GrapeCity.Documents.Excel.Examples.Features.Worksheets
{
    public class UnprotectWorksheet : ExampleBase
    {
        public override void Execute(GrapeCity.Documents.Excel.Workbook workbook)
        {
            IWorksheet worksheet = workbook.Worksheets[0];

            //protect worksheet, allow insert column.
            worksheet.Protection = true;
            worksheet.ProtectionSettings.AllowInsertingColumns = true;

            //Unprotect worksheet.
            worksheet.Protection = false;
        }

        public override bool ShowViewer
        {
            get
            {
                return false;
            }
        }
    }
}

File: Examples.VB/Features/Worksheets/UnprotectWorksheet.vb
Namespace Features.Worksheets
    Public Class UnprotectWorksheet
        Inherits ExampleBase
        Public Overrides Sub Execute(workbook As Excel.Workbook)
            Dim worksheet As IWorksheet = workbook.Worksheets(0)

            'protect worksheet, allow insert column.
            worksheet.Protection = True
            worksheet.ProtectionSettings.AllowInsertingColumns = True

            'Unprotect worksheet.
            worksheet.Protection = False
        End Sub
        Public Overrides ReadOnly Property ShowViewer As Boolean
            Get
                Return False
            End Get
        End Property
    End Class
End Namespace
